### PR TITLE
ci: update build

### PR DESCRIPTION
**Pull request by Ylarod**
1. update kernel ci
2. push ksud builds to `gh-ksud` branch
3. fix manager build

### CODE REVIEW

File: .github/workflows/build-ksud.yml [hunk 1]
```diff
@@ -19,4 +19,32 @@ jobs:
...
+  push-to-github:
+    needs: build
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download artifacts
+      uses: actions/download-artifact@v3
+    - name: Push to GitHub
+      run: |
+        BRANCH=gh-ksud
```
Comment (tiann) [inline .github/workflows/build-ksud.yml:32]: we shouldn't mess this pr with another pr

File: .github/workflows/gki-kernel.yml [hunk 2]
```diff
@@ -155,6 +113,7 @@ jobs:
           cd $GITHUB_WORKSPACE
           echo "[+] Build embed_ksud.c"
           python3 ./KernelSU/scripts/bin2c.py ./aarch64-linux-android/release/ksud ksud > ./KernelSU/kernel/embed_ksud.c
+          touch ./KernelSU/kernel/embed_ksud.lock.h
```
Comment (tiann) [inline .github/workflows/gki-kernel.yml:116]: don't mess it up

File: .github/workflows/gki-kernel.yml [hunk 3]
```diff
@@ -168,100 +127,20 @@ jobs:
...
-          path: ${{ env.OUTPUT_BOOT_GZ }}
-
-      - name: Upload boot-lz4_lg.img
-        uses: actions/upload-artifact@v3
-        with:
-          name: ${{ inputs.version_name }}-boot-lz4_lg.img
-          path: ${{ env.OUTPUT_BOOT_LZ4_LG }}
+          mkdir output
+          cp $OUTDIR/Image ./output/
+          cp $OUTDIR/Image.lz4 ./output/
```
Comment (tiann) [inline .github/workflows/gki-kernel.yml:140]: is this lz4 or lz4_legacy?

**Review by tiann — CHANGES_REQUESTED** [review]

File: .github/workflows/gki-kernel.yml (hunk 3 above)
Comment (Ylarod) [reply .github/workflows/gki-kernel.yml:140]: lz4_legacy in a13
lz4_lg in a12

File: .github/scripts/build_a12.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	echo "[+] Unpack prebuilt boot.img"
+	BOOT_IMG=$(find . -maxdepth 1 -name "boot*.img")
+	$UNPACK_BOOTIMG --boot_img=$BOOT_IMG
+	rm $BOOT_IMG
+
+	echo "[+] Building Image.gz"
+	cat Image | $GZIP -n -f -9 > Image.gz
+
+	echo "[+] Building boot.img"
+	$MKBOOTIMG --header_version 4 --kernel Image --output boot.img --ramdisk out/ramdisk --os_version 12.0.0 --os_patch_level 2021-11
```
[the remark this answers is not in the document]
Comment (tiann) [reply .github/scripts/build_a12.sh:21]: do we need to `--os_patch_level $PATCH_LEVEL?

File: .github/scripts/build_a12.sh [hunk 5]
```diff
@@ -0,0 +1,51 @@
+build_from_image(){
+	export TITLE=kernel-aarch64-$(echo $1 | sed 's/Image-//g')
+	echo "[+] title: $TITLE"
+
+	export PATCH_LEVEL=$(echo $1 | awk -F_ '{ print $2}')
+	echo "[+] patch level: $PATCH_LEVEL"
+
+	echo "[+] Download prebuilt ramdisk"
+	curl -Lo gki-kernel.zip https://dl.google.com/android/gki/gki-certified-boot-android12-5.10-$(PATCH_LEVEL)_r1.zip
```
Comment (tiann) [inline .github/scripts/build_a12.sh:9]: should be `$PATCH_LEVEL

File: .github/scripts/build_a12.sh [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+	echo "[+] Building Image.gz"
+	cat Image | $GZIP -n -f -9 > Image.gz
+
+	echo "[+] Building boot.img"
+	$MKBOOTIMG --header_version 4 --kernel Image --output boot.img --ramdisk out/ramdisk --os_version 12.0.0 --os_patch_level 2021-11
+	$AVBTOOL add_hash_footer --partition_name boot --partition_size $((64*1024*1024)) --image boot.img --algorithm SHA256_RSA2048 --key ../kernel-build-tools/linux-x86/share/avb/testkey_rsa2048.pem
+
+	echo "[+] Building boot-gz.img"
+	$MKBOOTIMG --header_version 4 --kernel Image.gz --output boot-gz.img --ramdisk out/ramdisk --os_version 12.0.0 --os_patch_level 2021-11
```
Comment (tiann) [inline .github/scripts/build_a12.sh:25]: patch_level

File: .github/scripts/build_a12.sh [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+	$AVBTOOL add_hash_footer --partition_name boot --partition_size $((64*1024*1024)) --image boot-gz.img --algorithm SHA256_RSA2048 --key ../kernel-build-tools/linux-x86/share/avb/testkey_rsa2048.pem
+
+	echo "[+] Building boot-lz4.img"
+	$MKBOOTIMG --header_version 4 --kernel Image.lz4 --output boot-lz4.img --ramdisk out/ramdisk --os_version 12.0.0 --os_patch_level 2021-11
```
Comment (tiann) [inline .github/scripts/build_a12.sh:29]: patch_level

**Review by tiann — CHANGES_REQUESTED** [review]

File: .github/scripts/build_a12.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (Ylarod) [reply .github/scripts/build_a12.sh:21]: forget to change

File: .github/scripts/build_a12.sh (hunk 5 above)
Comment (Ylarod) [reply .github/scripts/build_a12.sh:9]: $PATCH_LEVEL will add _r1 to it

Comment (tiann) [reply .github/scripts/build_a12.sh]: `${PATCH_LEVEL}`? `$(PATCH_LEVEL) is a subprocess call

**Review by tiann — APPROVED** [review]

**Pull request merged**